lib/sinatra/base.rb

```diff
@@ -465,7 +465,7 @@ def cache_control(*values)
       values.map! { |value| value.to_s.tr('_','-') }
       hash.each do |key, value|
         key = key.to_s.tr('_', '-')
-        value = value.to_i if key == "max-age"
+        value = value.to_i if /max-?age/ =~ key
```

Comment:
Elegant solution :+1: But this also work when there are typos in the maxage header. For example, `:s_maxages` or `s-max-age`. Do you think that will create confusion?